### PR TITLE
Fix/donation settings vs billing fields

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

In Reader Revenue Wizard, saving the donation settings updates the donation products. However, if a user does not wish to use the products, but wants to update the "Billing Fields" section, they'll get the donation products anyway. This PR fixes it so that donation settings and billing fields are in separate sections:

| Before | After | 
|--------------|-----------|
| image | image |

@Automattic/newspack-design – since this UI is in the process of getting an overhaul, I'm not sure if much design attention is needed here. Please advise.

### How to test the changes in this Pull Request:

1. On `trunk`
1. Set up a site with Newspack as the RR platform
2. Remove the WC donation products
3. Update "Billing Fields" section and save the settings – observe the donation products have been restored
4. Switch to this branch, delete the products again, update "Billing Fields" section and observe the products have not been restored
5. Update the donation settings section, observe the products have been restored

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207128165866630

### CODE REVIEW

File: includes/class-donations.php
Comment (adekbadek) [inline includes/class-donations.php:435]: This was added in #1895, as a migration step. By now we can be sure everything was migrated.

**Comment by thomasguillot** [comment]
> since this UI is in the process of getting an overhaul, I'm not sure if much design attention is needed here. Please advise.

I think this is fine for now 👍

**Review by dkoo — APPROVED** [review]
@adekbadek @thomasguillot This is working well from a functional standpoint, but the UI is really bothering me now with the separation into two sections. I know we said this UI will be undergoing more extensive changes in the near future, but what about doing a very minor visual refactor such as wrapping each section in an `ActionCard` with its own Save button, which is a pattern we've used in other dashboard pages? #3156 is my proposal—feel free to reject or make changes to that PR.

I'm going to approve this PR for now, but if #3156 looks like an incremental improvement over the current UI, please merge that one before merging this PR into `trunk`.

**Pull request merged**
